src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -40,6 +40,13 @@ namespace Microsoft.CodeAnalysis.LanguageServer;
 /// </remarks>
 internal class LspWorkspaceManager : IDocumentChangeTracker, IDisposable
 {
+    /// <summary>
+    /// Indicates whether the LSP solution has changed in a non-tracked document context.
+    /// 
+    /// <b>IMPORTANT:</b> Implementations of this event handler should not do any synchronous work since this will block.
```

Comment:
```suggestion
    /// <b>IMPORTANT:</b> Implementations of this event handler should do as little synchronous work as possible since this will block.
```